### ISSUE TITLE
re-installing lua builder fails

### ISSUE THREAD

**Issue by waleedhammam** [threefoldtech/jumpscaleX_builders#62]
- on  a fresh container, threebotserver starting and everything is ok

- try to reinstall lua builder

- in kosmos `j.builders.runtimes.lua.install(reset=True)`

```bash
Configuring LuaRocks version 3.1.3...

Lua version detected: 5.1
Lua interpreter found: /sandbox/openresty/luajit/bin/luajit
lua.h found: /sandbox/openresty/luajit/include/luajit-2.1/lua.h
unzip found in PATH: /bin

Done configuring.

LuaRocks will be installed at......: /sandbox/openresty/luarocks
LuaRocks will install rocks at.....: /sandbox/openresty/luarocks
LuaRocks configuration directory...: /sandbox/openresty/luarocks/etc/luarocks
Using Lua from.....................: /sandbox/openresty/luajit

* Type make and make install:
  to install to /sandbox/openresty/luarocks as usual.
* Type make bootstrap:
  to install LuaRocks into /sandbox/openresty/luarocks as a rock.

+ make build
mkdir -p "build"
rm -f src/luarocks/core/hardcoded.lua
echo "#!/bin/sh" > luarocks
echo "unset LUA_PATH LUA_PATH_5_2 LUA_PATH_5_3 LUA_PATH_5_4" >> luarocks
echo 'LUAROCKS_SYSCONFDIR="/sandbox/openresty/luarocks/etc/luarocks" LUA_PATH="/tmp/builders/lua/luarocks/src/?.lua;;" exec "/sandbox/openresty/luajit/bin/luajit" "/tmp/builders/lua/luarocks/src/bin/luarocks" --project-tree="/tmp/builders/lua/luarocks/lua_modules" "$@"' >> luarocks
chmod +rx ./luarocks
./luarocks init
/sandbox/openresty/luajit/bin/luajit: /tmp/builders/lua/luarocks/src/luarocks/deps.lua:13: module 'luarocks.build.builtin' not found:No LuaRocks module found for luarocks.build.builtin
	no field package.preload['luarocks.build.builtin']
	no file '/tmp/builders/lua/luarocks/src/luarocks/build/builtin.lua'
	no file './luarocks/build/builtin.lua'
	no file '/sandbox/openresty/luajit/share/luajit-2.1.0-beta3/luarocks/build/builtin.lua'
	no file '/usr/local/share/lua/5.1/luarocks/build/builtin.lua'
	no file '/usr/local/share/lua/5.1/luarocks/build/builtin/init.lua'
	no file '/sandbox/openresty/luajit/share/lua/5.1/luarocks/build/builtin.lua'
	no file '/sandbox/openresty/luajit/share/lua/5.1/luarocks/build/builtin/init.lua'
	no file '/sandbox/openresty/lualib/luarocks/build/builtin.so'
	no file './luarocks/build/builtin.so'
	no file '/sandbox/openresty/lualib/luarocks.so'
	no file './luarocks.so'
stack traceback:
	[C]: in function 'require'
	/tmp/builders/lua/luarocks/src/luarocks/deps.lua:13: in main chunk
	[C]: in function 'require'
	/tmp/builders/lua/luarocks/src/luarocks/cmd.lua:10: in main chunk
	[C]: in function 'require'
	/tmp/builders/lua/luarocks/src/bin/luarocks:7: in main chunk
	[C]: at 0x55980c13e2a0
make: *** [GNUmakefile:55: luarocks] Error 1
Sat 11 09:45:27 core/InstallTools.py -1428 - _execute                           : EXCEPTION: 

    Could not execute: 'bash /tmp/builder_lua.sh'stdout:

        Configuring LuaRocks version 3.1.3...

        Lua version detected: 5.1
        Lua interpreter found: /sandbox/openresty/luajit/bin/luajit
        lua.h found: /sandbox/openresty/luajit/include/luajit-2.1/lua.h
        unzip found in PATH: /bin

        Done configuring.

        LuaRocks will be installed at......: /sandbox/openresty/luarocks
        LuaRocks will install rocks at.....: /sandbox/openresty/luarocks
        LuaRocks configuration directory...: /sandbox/openresty/luarocks/etc/luarocks
        Using Lua from.....................: /sandbox/openresty/luajit

        * Type make and make install:
          to install to /sandbox/openresty/luarocks as usual.
        * Type make bootstrap:
          to install LuaRocks into /sandbox/openresty/luarocks as a rock.

        mkdir -p "build"
        rm -f src/luarocks/core/hardcoded.lua
        echo "#!/bin/sh" > luarocks
        echo "unset LUA_PATH LUA_PATH_5_2 LUA_PATH_5_3 LUA_PATH_5_4" >> luarocks
        echo 'LUAROCKS_SYSCONFDIR="/sandbox/openresty/luarocks/etc/luarocks"
        LUA_PATH="/tmp/builders/lua/luarocks/src/?.lua;;"
            exec "/sandbox/openresty/luajit/bin/luajit" "/tmp/builders/lua/luarocks/src/bin/luarocks" --project-
            tree="/tmp/builders/lua/luarocks/lua_modules" "$@"' >> luarocks
        chmod +rx ./luarocks
        ./luarocks init

    stderr:
        + cd /tmp/
        + . /tmp/builders/lua/env.sh
        ++ export LIBRARY_PATH=/lib/x86_64-linux-gnu:/lib:/usr/local/lib:/usr/lib
        ++ LIBRARY_PATH=/lib/x86_64-linux-gnu:/lib:/usr/local/lib:/usr/lib
        ++ export LD_LIBRARY_PATH=/lib/x86_64-linux-gnu:/lib:/usr/local/lib:/usr/lib
        ++ LD_LIBRARY_PATH=/lib/x86_64-linux-gnu:/lib:/usr/local/lib:/usr/lib
        ++ export 'LDFLAGS=-L/lib/x86_64-linux-gnu -L/lib -L/usr/local/lib -L/usr/lib'
        ++ LDFLAGS='-L/lib/x86_64-linux-gnu -L/lib -L/usr/local/lib -L/usr/lib'
        ++ export CPPPATH=/usr/include
        ++ CPPPATH=/usr/include
        ++ export CPPFLAGS=-I/usr/include
        ++ CPPFLAGS=-I/usr/include
        ++ export PS1=PYTHONBUILDENV:
        ++ PS1=PYTHONBUILDENV:
        ++ export PYTHONHTTPSVERIFY=0
        ++ PYTHONHTTPSVERIFY=0
        ++ export LC_ALL=en_US.UTF-8
        ++ LC_ALL=en_US.UTF-8
        ++ export LANG=en_US.UTF-8
        ++ LANG=en_US.UTF-8
        + cd /tmp/builders/lua/luarocks
        + ./configure --prefix=/sandbox/openresty/luarocks --with-lua=/sandbox/openresty/luajit
        + make build
        /sandbox/openresty/luajit/bin/luajit: /tmp/builders/lua/luarocks/src/luarocks/deps.lua:13: module
            'luarocks.build.builtin' not found:No LuaRocks module found for luarocks.build.builtin
                no field package.preload['luarocks.build.builtin']
                no file '/tmp/builders/lua/luarocks/src/luarocks/build/builtin.lua'
                no file './luarocks/build/builtin.lua'
                no file '/sandbox/openresty/luajit/share/luajit-2.1.0-beta3/luarocks/build/builtin.lua'
                no file '/usr/local/share/lua/5.1/luarocks/build/builtin.lua'
                no file '/usr/local/share/lua/5.1/luarocks/build/builtin/init.lua'
                no file '/sandbox/openresty/luajit/share/lua/5.1/luarocks/build/builtin.lua'
                no file '/sandbox/openresty/luajit/share/lua/5.1/luarocks/build/builtin/init.lua'
                no file '/sandbox/openresty/lualib/luarocks/build/builtin.so'
                no file './luarocks/build/builtin.so'
                no file '/sandbox/openresty/lualib/luarocks.so'
                no file './luarocks.so'
        stack traceback:
                [C]: in function 'require'
                /tmp/builders/lua/luarocks/src/luarocks/deps.lua:13: in main chunk
                [C]: in function 'require'
                /tmp/builders/lua/luarocks/src/luarocks/cmd.lua:10: in main chunk
                [C]: in function 'require'
                /tmp/builders/lua/luarocks/src/bin/luarocks:7: in main chunk
                [C]: at 0x55980c13e2a0
        make: *** [GNUmakefile:55: luarocks] Error 1
--TRACEBACK------------------
<stdin> in <module>
    1    
/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/BuilderBaseClass.py in wrapper_action
    158    builder.build(reset=reset)
/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/BuilderBaseClass.py in wrapper_action
    173    res = self.apply_method(func, kwargs)
/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/BuilderBaseClass.py in apply_method
    93    return func(**kwargs)
/sandbox/lib/jumpscale/JumpscaleBuilders/runtimes/BuilderLua.py in build
    50    self._execute(C, showout=True)
/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/BuilderBaseClass.py in _execute
    385    retry=retry,
/sandbox/lib/jumpscale/Jumpscale/sal/process/SystemProcess.py in execute
    117    retry=retry,
/sandbox/lib/jumpscale/Jumpscale/core/InstallTools.py in execute
    2669    errormsg=errormsg,
/sandbox/lib/jumpscale/Jumpscale/core/InstallTools.py in _execute
    1428    raise Tools.exceptions.Base(msg)
-----------------------------

Sat 11 09:45:27 core/InstallTools.py -1428 - _execute                           : EXCEPTION: 

    Could not execute: 'bash /tmp/builder_lua.sh'stdout:

        Configuring LuaRocks version 3.1.3...

        Lua version detected: 5.1
        Lua interpreter found: /sandbox/openresty/luajit/bin/luajit
        lua.h found: /sandbox/openresty/luajit/include/luajit-2.1/lua.h
        unzip found in PATH: /bin

        Done configuring.

        LuaRocks will be installed at......: /sandbox/openresty/luarocks
        LuaRocks will install rocks at.....: /sandbox/openresty/luarocks
        LuaRocks configuration directory...: /sandbox/openresty/luarocks/etc/luarocks
        Using Lua from.....................: /sandbox/openresty/luajit

        * Type make and make install:
          to install to /sandbox/openresty/luarocks as usual.
        * Type make bootstrap:
          to install LuaRocks into /sandbox/openresty/luarocks as a rock.

        mkdir -p "build"
        rm -f src/luarocks/core/hardcoded.lua
        echo "#!/bin/sh" > luarocks
        echo "unset LUA_PATH LUA_PATH_5_2 LUA_PATH_5_3 LUA_PATH_5_4" >> luarocks
        echo 'LUAROCKS_SYSCONFDIR="/sandbox/openresty/luarocks/etc/luarocks"
        LUA_PATH="/tmp/builders/lua/luarocks/src/?.lua;;"
            exec "/sandbox/openresty/luajit/bin/luajit" "/tmp/builders/lua/luarocks/src/bin/luarocks" --project-
            tree="/tmp/builders/lua/luarocks/lua_modules" "$@"' >> luarocks
        chmod +rx ./luarocks
        ./luarocks init

    stderr:
        + cd /tmp/
        + . /tmp/builders/lua/env.sh
        ++ export LIBRARY_PATH=/lib/x86_64-linux-gnu:/lib:/usr/local/lib:/usr/lib
        ++ LIBRARY_PATH=/lib/x86_64-linux-gnu:/lib:/usr/local/lib:/usr/lib
        ++ export LD_LIBRARY_PATH=/lib/x86_64-linux-gnu:/lib:/usr/local/lib:/usr/lib
        ++ LD_LIBRARY_PATH=/lib/x86_64-linux-gnu:/lib:/usr/local/lib:/usr/lib
        ++ export 'LDFLAGS=-L/lib/x86_64-linux-gnu -L/lib -L/usr/local/lib -L/usr/lib'
        ++ LDFLAGS='-L/lib/x86_64-linux-gnu -L/lib -L/usr/local/lib -L/usr/lib'
        ++ export CPPPATH=/usr/include
        ++ CPPPATH=/usr/include
        ++ export CPPFLAGS=-I/usr/include
        ++ CPPFLAGS=-I/usr/include
        ++ export PS1=PYTHONBUILDENV:
        ++ PS1=PYTHONBUILDENV:
        ++ export PYTHONHTTPSVERIFY=0
        ++ PYTHONHTTPSVERIFY=0
        ++ export LC_ALL=en_US.UTF-8
        ++ LC_ALL=en_US.UTF-8
        ++ export LANG=en_US.UTF-8
        ++ LANG=en_US.UTF-8
        + cd /tmp/builders/lua/luarocks
        + ./configure --prefix=/sandbox/openresty/luarocks --with-lua=/sandbox/openresty/luajit
        + make build
        /sandbox/openresty/luajit/bin/luajit: /tmp/builders/lua/luarocks/src/luarocks/deps.lua:13: module
            'luarocks.build.builtin' not found:No LuaRocks module found for luarocks.build.builtin
                no field package.preload['luarocks.build.builtin']
                no file '/tmp/builders/lua/luarocks/src/luarocks/build/builtin.lua'
                no file './luarocks/build/builtin.lua'
                no file '/sandbox/openresty/luajit/share/luajit-2.1.0-beta3/luarocks/build/builtin.lua'
                no file '/usr/local/share/lua/5.1/luarocks/build/builtin.lua'
                no file '/usr/local/share/lua/5.1/luarocks/build/builtin/init.lua'
                no file '/sandbox/openresty/luajit/share/lua/5.1/luarocks/build/builtin.lua'
                no file '/sandbox/openresty/luajit/share/lua/5.1/luarocks/build/builtin/init.lua'
                no file '/sandbox/openresty/lualib/luarocks/build/builtin.so'
                no file './luarocks/build/builtin.so'
                no file '/sandbox/openresty/lualib/luarocks.so'
                no file './luarocks.so'
        stack traceback:
                [C]: in function 'require'
                /tmp/builders/lua/luarocks/src/luarocks/deps.lua:13: in main chunk
                [C]: in function 'require'
                /tmp/builders/lua/luarocks/src/luarocks/cmd.lua:10: in main chunk
                [C]: in function 'require'
                /tmp/builders/lua/luarocks/src/bin/luarocks:7: in main chunk
                [C]: at 0x55980c13e2a0
        make: *** [GNUmakefile:55: luarocks] Error 1
--TRACEBACK------------------
<stdin> in <module>
    1    
/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/BuilderBaseClass.py in wrapper_action
    158    builder.build(reset=reset)
/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/BuilderBaseClass.py in wrapper_action
    173    res = self.apply_method(func, kwargs)
/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/BuilderBaseClass.py in apply_method
    93    return func(**kwargs)
/sandbox/lib/jumpscale/JumpscaleBuilders/runtimes/BuilderLua.py in build
    50    self._execute(C, showout=True)
/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/BuilderBaseClass.py in _execute
    385    retry=retry,
/sandbox/lib/jumpscale/Jumpscale/sal/process/SystemProcess.py in execute
    117    retry=retry,
/sandbox/lib/jumpscale/Jumpscale/core/InstallTools.py in execute
    2669    errormsg=errormsg,
/sandbox/lib/jumpscale/Jumpscale/core/InstallTools.py in _execute
    1428    raise Tools.exceptions.Base(msg)
-----------------------------


Could not execute: 'bash /tmp/builder_lua.sh'stdout:

    ?[1;36mConfiguring LuaRocks version 3.1.3...?[0m

    Lua version detected: ?[1;32m5.1?[0m
    Lua interpreter found: ?[1;32m/sandbox/openresty/luajit/bin/luajit?[0m
    lua.h found: ?[1;32m/sandbox/openresty/luajit/include/luajit-2.1/lua.h?[0m
    unzip found in PATH: ?[1;32m/bin?[0m

    ?[1;36mDone configuring.?[0m

    LuaRocks will be installed at......: ?[1;32m/sandbox/openresty/luarocks?[0m
    LuaRocks will install rocks at.....: ?[1;32m/sandbox/openresty/luarocks?[0m
    LuaRocks configuration directory...: ?[1;32m/sandbox/openresty/luarocks/etc/luarocks?[0m
    Using Lua from.....................: ?[1;32m/sandbox/openresty/luajit?[0m

    * Type ?[1mmake?[0m and ?[1mmake install?[0m:
      to install to /sandbox/openresty/luarocks as usual.
    * Type ?[1mmake bootstrap?[0m:
      to install LuaRocks into /sandbox/openresty/luarocks as a rock.

    mkdir -p "build"
    rm -f src/luarocks/core/hardcoded.lua
    echo "#!/bin/sh" > luarocks
    echo "unset LUA_PATH LUA_PATH_5_2 LUA_PATH_5_3 LUA_PATH_5_4" >> luarocks
    echo 'LUAROCKS_SYSCONFDIR="/sandbox/openresty/luarocks/etc/luarocks" LUA_PATH="/tmp/builders/lua/luarocks/src/?.lua;;"
        exec "/sandbox/openresty/luajit/bin/luajit" "/tmp/builders/lua/luarocks/src/bin/luarocks" --project-
        tree="/tmp/builders/lua/luarocks/lua_modules" "$@"' >> luarocks
    chmod +rx ./luarocks
    ./luarocks init

stderr:
    + cd /tmp/
    + . /tmp/builders/lua/env.sh
    ++ export LIBRARY_PATH=/lib/x86_64-linux-gnu:/lib:/usr/local/lib:/usr/lib
    ++ LIBRARY_PATH=/lib/x86_64-linux-gnu:/lib:/usr/local/lib:/usr/lib
    ++ export LD_LIBRARY_PATH=/lib/x86_64-linux-gnu:/lib:/usr/local/lib:/usr/lib
    ++ LD_LIBRARY_PATH=/lib/x86_64-linux-gnu:/lib:/usr/local/lib:/usr/lib
    ++ export 'LDFLAGS=-L/lib/x86_64-linux-gnu -L/lib -L/usr/local/lib -L/usr/lib'
    ++ LDFLAGS='-L/lib/x86_64-linux-gnu -L/lib -L/usr/local/lib -L/usr/lib'
    ++ export CPPPATH=/usr/include
    ++ CPPPATH=/usr/include
    ++ export CPPFLAGS=-I/usr/include
    ++ CPPFLAGS=-I/usr/include
    ++ export PS1=PYTHONBUILDENV:
    ++ PS1=PYTHONBUILDENV:
    ++ export PYTHONHTTPSVERIFY=0
    ++ PYTHONHTTPSVERIFY=0
    ++ export LC_ALL=en_US.UTF-8
    ++ LC_ALL=en_US.UTF-8
    ++ export LANG=en_US.UTF-8
    ++ LANG=en_US.UTF-8
    + cd /tmp/builders/lua/luarocks
    + ./configure --prefix=/sandbox/openresty/luarocks --with-lua=/sandbox/openresty/luajit
    + make build
    /sandbox/openresty/luajit/bin/luajit: /tmp/builders/lua/luarocks/src/luarocks/deps.lua:13: module
        'luarocks.build.builtin' not found:No LuaRocks module found for luarocks.build.builtin
            no field package.preload['luarocks.build.builtin']
            no file '/tmp/builders/lua/luarocks/src/luarocks/build/builtin.lua'
            no file './luarocks/build/builtin.lua'
            no file '/sandbox/openresty/luajit/share/luajit-2.1.0-beta3/luarocks/build/builtin.lua'
            no file '/usr/local/share/lua/5.1/luarocks/build/builtin.lua'
            no file '/usr/local/share/lua/5.1/luarocks/build/builtin/init.lua'
            no file '/sandbox/openresty/luajit/share/lua/5.1/luarocks/build/builtin.lua'
            no file '/sandbox/openresty/luajit/share/lua/5.1/luarocks/build/builtin/init.lua'
            no file '/sandbox/openresty/lualib/luarocks/build/builtin.so'
            no file './luarocks/build/builtin.so'
            no file '/sandbox/openresty/lualib/luarocks.so'
            no file './luarocks.so'
    stack traceback:
            [C]: in function 'require'
            /tmp/builders/lua/luarocks/src/luarocks/deps.lua:13: in main chunk
            [C]: in function 'require'
            /tmp/builders/lua/luarocks/src/luarocks/cmd.lua:10: in main chunk
            [C]: in function 'require'
            /tmp/builders/lua/luarocks/src/bin/luarocks:7: in main chunk
            [C]: at 0x55980c13e2a0
    make: *** [GNUmakefile:55: luarocks] Error 1

```

**Comment by xmonader** [comment]
# using git bisect against 10.4-rc4 and HEAD of unstable  

```
3BOTDEVEL:3bot:jumpscaleX_core: cat /tmp/test_lua.sh 
#!/bin/bash
bcdb delete --all -f
kosmos -p 'j.builders.runtimes.lua.build(reset=True)'
```

```
3BOTDEVEL:3bot:jumpscaleX_core: git bisect start HEAD 9fd3f5394d683a21db57d48e83666bdc9a896ad6 --
```

```
3BOTDEVEL:3bot:jumpscaleX_core: git bisect run /tmp/test_lua.sh


935df2fa91edafcf681d71a79cf2864dccb3bbba is the first bad commit
commit 935df2fa91edafcf681d71a79cf2864dccb3bbba
Author: kristof <kristof@incubaid.com>
Date:   Fri Apr 10 08:10:11 2020 +0200

    jsx building seems to work now, more tests needed

:040000 040000 836e4333dc5b3539c029f54231662cc32bd0f404 464d1dafda2a03a478495eec6edb7d3e78f4a07b M  JumpscaleCore
:040000 040000 b6a12bd6a834ca2b6517ba4b05dd084f32effa0f fc67978f95b0cf4a7b0cd0e6433a512895dd15de M  install
bisect run success
```
problem is the addition of `build, dist` to default_ignore_dirs in sal/fs https://github.com/threefoldtech/jumpscaleX_core/commit/935df2fa91edafcf681d71a79cf2864dccb3bbba#diff-142bf38cc98ec0c18e38f4b0acf1aa2aR185 here

**Comment by xmonader** [comment]
fixed in threefoldtech/jumpscaleX_core@66c861b77c2e33c4fc33e4899a06ca7b291a18c9
